Weather/src/app/app.module.ts

```diff
@@ -0,0 +1,37 @@
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { AppRoutingModule } from './app-routing.module'; 
+import { HttpClientModule } from '@angular/common/http';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { MatSidenavModule } from '@angular/material/sidenav';
+import { MatIconModule } from '@angular/material/icon';
+import { MatListModule } from '@angular/material/list';
+import { MatButtonModule } from '@angular/material/button';
+import { GeolocationService } from './services/geolocation/geolocation.service';
+
+
+
+import { AppComponent } from './app.component';
+import { WeatherComponent } from './weather/weather.component';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+@NgModule({
+  declarations: [
+    AppComponent,
+    WeatherComponent,
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    HttpClientModule,
+    BrowserAnimationsModule,
+    MatToolbarModule,
+    MatIconModule,
+    MatSidenavModule,
+    MatListModule,
+    MatButtonModule 
+  ],
+  providers: [GeolocationService],
```

Comment:
If your service is provided as 'root' it's unnecessary.











`providedIn: 'root' is the easiest and most efficient way to provide services since Angular 6:

The service will be available application wide as a singleton with no need to add it to a module's providers array (like Angular <= 5).
If the service is only used within a lazy loaded module it will be lazy loaded with that module
If it is never used it will not be contained in the build (tree shaked).
For further informations consider reading the [documentation](https://angular.io/guide/providers#providing-a-service) and [NgModule FAQs](https://angular.io/guide/ngmodule-faq)

Btw:

If you don't want a application-wide singleton use the provider's array of a component instead.
If you want to limit the scope so no other developer will ever use your service outside of a particular module, use the providers array of NgModule instead.`